modules/caddyhttp/matchers.go

```diff
@@ -909,25 +914,51 @@ func (m MatchRemoteIP) getClientIP(r *http.Request) (net.IP, error) {
 	if err != nil {
 		ipStr = remote // OK; probably didn't have a port
 	}
+	// Some IPv6-Adresses can contain zone identifiers at the end,
+	// which are separated with "%"
+	if strings.Contains(ipStr, "%") {
+		zone_id = strings.Split(ipStr, "%")[1]
+		ipStr = strings.Split(ipStr, "%")[0]
+	}
 	ip := net.ParseIP(ipStr)
 	if ip == nil {
-		return nil, fmt.Errorf("invalid client IP address: %s", ipStr)
+		return nil, zone_id, fmt.Errorf("invalid client IP address: %s", ipStr)
 	}
-	return ip, nil
+	return ip, zone_id, nil
 }
 
 // Match returns true if r matches m.
 func (m MatchRemoteIP) Match(r *http.Request) bool {
-	clientIP, err := m.getClientIP(r)
+	clientIP, zone_id_got, err := m.getClientIP(r)
+	zoneFilter := true
 	if err != nil {
 		m.logger.Error("getting client IP", zap.Error(err))
 		return false
 	}
-	for _, ipRange := range m.cidrs {
+	for i, ipRange := range m.cidrs {
+
 		if ipRange.Contains(clientIP) {
-			return true
+			// Assign the right ip string for zone_id handling.
+			zone_id_def := m.Ranges[i]
+			// Check if the filter ips contain zone identifiers.
+			if strings.Contains(zone_id_def, "%") {
+				zone_id_def = strings.Split(zone_id_def, "%")[1]
+				// Check if the incomming zone_id is that one in filter.
+				// Assign true or false for the handling of multible ips with or without zone_ids in Filter
+				// and handling of the "zone_id not match" info.
+				if zone_id_def == zone_id_got {
+					return true
+				} else {
+					zoneFilter = false
+				}
+			} else {
+				return true
+			}
 		}
 	}
+	if zoneFilter == false {
+		m.logger.Info("zone_id from remote Client, don't matches that one in remote_ip filter.")
+	}
```

Comment:
I don't think we need this log. It might be pretty noisy. It doesn't tell the user anything that they couldn't figure out from looking at the access logs. We _could_ log it at `Debug` level which would make it possible to reveal if they're trying to figure out if their matcher is broken, though.

Also the wording is somewhat awkward, I might write it like this:

```suggestion
	if !zoneFilter {
		m.logger.Debug("zone ID from remote '%s' did not match", zoneID)
	}
```

Reply:
The first park makes sense, but I can't change the error text itself.
```
cannot use zoneID (type string) as type zapcore.Field in argument to m.logger.Debug
```

Reply:
Oh right, that's a zap logger and not fmt, need to do this instead: `m.logger.Debug("zone ID from remote did not match", zap.String("zone", zoneID))`